frontend/packages/console-app/src/components/user-preferences/theme/ThemeField.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { THEME_LOCAL_STORAGE_KEY } from './const';
+import { usePreferredTheme } from './usePreferredTheme';
+
+const ThemeField: React.FC = () => {
+  const { t } = useTranslation();
+  const [preferredTheme, setPreferredTheme, preferredThemeLoaded] = usePreferredTheme();
+  const [isThemeOpen, setIsThemeOpen] = React.useState<boolean>(false);
+
+  const ThemeTypes = {
+    systemDefault: t('console-app~System Default'),
+    light: t('console-app~Light'),
+    dark: t('console-app~Dark'),
+  };
+
+  const themeSelectOptions: JSX.Element[] = React.useMemo(
+    () =>
+      Object.keys(ThemeTypes).map((themeOption) => (
+        <SelectOption key={themeOption} value={themeOption}>
+          {ThemeTypes[themeOption]}
+        </SelectOption>
+      )),
+    [ThemeTypes],
+  );
+
+  const onThemeToggle = React.useCallback((isOpen: boolean) => setIsThemeOpen(isOpen), []);
+
+  const onThemeSelect = React.useCallback(
+    (_, selection: string) => {
+      setPreferredTheme(selection);
+      setIsThemeOpen(false);
+
+      const root = document.getElementsByTagName('html')[0];
+
+      if (selection === 'dark') {
+        root.classList.add('pf-theme-dark');
+      } else if (selection === 'systemDefault') {
+        if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
+          root.classList.add('pf-theme-dark');
+        } else {
+          root.classList.remove('pf-theme-dark');
+        }
+      } else {
+        root.classList.remove('pf-theme-dark');
+      }
```

Comment:
Preference components should deal only with setting the preference and not side effects related to those preferences. Let other parts of the app react to preference changes.

Note that this custom component shouldn't be necessary for the theme preferences.